public/scripts/index.js

```diff
@@ -0,0 +1,171 @@
+let canvas = document.getElementById('canvas');
+let context = canvas.getContext("2d");
+const cellSize = 10;
+let body = document.getElementById('body').addEventListener('click', handleMouseClick);
+let status;
+let value = 0;
+let rows = document.getElementById("rows").value;
+let cols = document.getElementById("columns").value;
+
+window.addEventListener('DOMContentLoaded', init);
+
+function init(){
+    console.log('Hello from init');
+    let submitBtn = document.getElementById('submit-button');
+    submitBtn.addEventListener('click', getInputValue);
+    let simulateBtn = document.getElementById('simulate-button');
+    simulateBtn.addEventListener('click', simulate);
+    let clearBtn1 = document.getElementById('clear-button');
+    clearBtn1.addEventListener('click', clear);
+    let drawPresetBtn = document.getElementById('drawPreset-button');
+    drawPresetBtn.addEventListener('click', drawPreset);
+    let selectPresetBtn = document.getElementById('preset1');
+    selectPresetBtn.addEventListener('click', getPreset);
+    let animateBtn = document.getElementById('animate');
+    animateBtn.addEventListener('click', animate);
+}
+function handleMouseClick(event){
+    const boundingRect = canvas.getBoundingClientRect();
+    let point = calcClickCoordinates(event.clientX, event.clientY, boundingRect.left, boundingRect.top, boundingRect.right, boundingRect.bottom);
+    if(point){
+        let point2 = calcCellCoordinates(point.x , point.y, cellSize);
+        if(currentFrame[point2.y][point2.x] === 0){
+            currentFrame[point2.y][point2.x] = 1;
+        }
+        else{
+            currentFrame[point2.y][point2.x] = 0;
+        }
+        draw(currentFrame, context, cellSize);
+        spanMouseX.innerHTML = 'Column ' + point2.x;
+        spanMouseY.innerHTML = 'Row ' + point2.y;
+    }
+    else {
+        spanMouseX.innerHTML = 'Out of bounds';
+        spanMouseY.innerHTML = 'Out of bounds';
+    }
+}
+function getInputValue() {
+    rows = document.getElementById("rows").value;
+    cols = document.getElementById("columns").value;
+
+    const h = rows * cellSize;
+    canvas.height = h;
+
+    const w = cols * cellSize;
+    canvas.width = w;
+
+    const arrH = h / cellSize;
+    const arrW = w / cellSize;
+
+    currentFrame = makeMatrix(arrH, arrW);
+    nextFrame = makeMatrix(arrH, arrW);
+    intermediateFrame = makeMatrix(arrH, arrW);
+
+    let context = canvas.getContext("2d");
+
+    drawBoard(w, h, context, cellSize);
+}
+
+function simulate(){
+    let intermediateFrame = getIntermediateFrame(currentFrame, rows, cols);
+    for (let i = 0; i < intermediateFrame.length; i++){
+        for (let j = 0; j < intermediateFrame[0].length; j++){
+                nextFrame[i][j] = calcNewCellState(currentFrame[i][j], intermediateFrame[i][j]);
+        }
+    }
+    currentFrame = nextFrame;
+    draw(currentFrame, context, cellSize);        
+}
+
+function drawPreset(){
```

Comment:
I suggest splitting this in to smaller functions. One function per preset. For example make a function called `drawAcorn` which knows how to draw an acorn and call it from `drawPreset`